dashboard/app/controllers/curriculum_tracking_pixel_controller.rb

```diff
@@ -14,30 +14,36 @@ def index
       # For ease of querying we attempt to parse the url into meaningful chunks.
       # EXAMPLE:
       # /csf-18/pre-express/11/
-      # es-mx/csf-1718/coursec/10/'
-      split_url = curriculum_page.split('/')
-      # ["", "csf-18", "pre-express", "11"]
+      # /es-mx/csf-1718/coursec/10/'
+      split_url = curriculum_page.split('/').reject(&:empty?)
+      # ["csf-18", "pre-express", "11"]
       # ["es-mx", "csf-1718", "coursec", "10"]
 
-      non_en = !!split_url[0].match(/\S{2}-\S{2}/)
-      locale = non_en ? split_url[0] : "en-us"
+      non_en = split_url[0].length == 5 && !!split_url[0].match(/\S{2}-\S{2}/)
 
-      if split_url.length > 1
+      if non_en
+        locale = split_url[0]
+        split_url.shift(1)
+      else
+        locale = "en-us"
+      end
```

Comment:
this whole block can be reduced to:
```suggestion
locale = non_en ? split_url.shift : 'en_us'
```